public/app/features/dimensions/editors/ResourceCards.tsx

```diff
@@ -0,0 +1,108 @@
+import React, { memo, CSSProperties } from 'react';
+import { areEqual, FixedSizeGrid as Grid } from 'react-window';
+import AutoSizer from 'react-virtualized-auto-sizer';
+import { GrafanaTheme2, SelectableValue } from '@grafana/data';
+import { useTheme2, stylesFactory } from '@grafana/ui';
+import SVG from 'react-inlinesvg';
+import { css } from '@emotion/css';
+
+interface CellProps {
+  columnIndex: number;
+  rowIndex: number;
+  style: CSSProperties;
+  data: any;
+}
+
+function Cell(props: CellProps) {
+  const { columnIndex, rowIndex, style, data } = props;
+  const { cards, columnCount, onChange, folder } = data;
+  const singleColumnIndex = columnIndex + rowIndex * columnCount;
+  const card = cards[singleColumnIndex];
+  const theme = useTheme2();
+  const styles = getStyles(theme, theme.colors.text.primary);
+
+  return (
+    <div style={style}>
+      {card && (
+        <div
+          key={card.value}
+          style={{
```

Comment:
use emotion here -- fixed style values should can be put into a class.  Only use explicit style when the values change often and based on data.  In this case each item in the search results has to parse all the CSS!